lib/utils/object_utils.js

```diff
@@ -45,6 +45,8 @@ function merge(object1, object2) {
 }
 
 function areEntitiesEqual(firstEntity, secondEntity) {
+  Object.keys(firstEntity).forEach(key => firstEntity[key] === undefined && delete firstEntity[key]);
```

Comment:
For clarity: 

```javascript
Object.keys(firstEntity).forEach(key => {
  if (!firstEntity[key]) {
    delete firstEntity[key];
  }
});
```

Reply:
Using `!firstEntity[key]` causes removing also `null`, `''` and `false` values which can lead to unexpected behaviour.  
I'd rather use `firstEntity[key] === undefined` in this suggested construct.

Reply:
true that, good point